packages/lesswrong/components/dialogues/DialoguesList.tsx

```diff
@@ -180,71 +278,60 @@ const DialoguesList = ({ classes }: { classes: ClassesType<typeof styles> }) =>
-                    checkId={checkId}
-                    isChecked={userIsChecked}
-                    isMatched={userIsMatched}
-                  />
-                </div>
-                <PostsItem2MetaInfo className={classes.dialogueMatchUsername}>
-                  <UsersName
-                    documentId={targetUser._id}
-                    simple={false}
-                  />
-                </PostsItem2MetaInfo>
-                <PostsItem2MetaInfo className={classes.dialogueMatchNote}>
-                  You've matched with this user
-                </PostsItem2MetaInfo>
-              </div>
-              <div className={classes.dialogueRightContainer}>
-                <div className={classes.dialogueMatchMessageButton}>
-                  <MessageButton
-                    targetUserId={targetUser._id}
-                    currentUser={currentUser}
-                    isMatched={userIsMatched}
-                  />
-                </div>
-                <div className={classes.dialogueMatchPreferencesButton}>
-                  <DialogueNextStepsButton
-                    isMatched={userIsMatched}
-                    checkId={checkId}
-                    targetUserId={targetUser._id}
-                    targetUserDisplayName={targetUser.displayName}
-                    currentUser={currentUser}
-                  />
-                </div>
-              </div>
-            </div>);
-          })}
-        </div>
+        {showReciprocityRecommendations && <div>
+          { (currentUser?.showMatches || currentUser?.showRecommendedPartners ) && recommendedDialoguePartnersRowPropsList.length > 0 &&
+            <div className={classes.explanatoryNoteBox}>
+              <Typography
+                component='span'
+                className={classes.explanatoryNote}
+                variant='body2'>
+                  {<div>
+                    Check a user you'd maybe dialogue with. They can't see your checks unless you match. If you match, you both get to enter topics and then choose whether to dialogue. (<a className={classes.link} href="https://www.lesswrong.com/posts/d65Ax6vbNgztBE8cy/new-lesswrong-feature-dialogue-matching">Learn more</a>)
+                    </div>}
+              </ Typography>
+            </div>}
+          {currentUser?.showMatches && matchRowPropsList?.map((rowProps, index) => (
+            <DialogueMatchRow key={index} rowProps={rowProps} classes={classes} showMatchNote={true} />
+          ))}
+          {currentUser?.showRecommendedPartners && recommendedDialoguePartnersRowPropsList?.map((rowProps, index) => (
+            <DialogueRecommendationRow key={index} rowProps={rowProps} showSuggestedTopics={showTopics} />
+          ))}
+        </div>}
       </AnalyticsContext>}
       
-      {dialoguePosts?.map((post, i: number) =>
+      {(!currentUser || currentUser?.showDialoguesList) && dialoguePosts?.map((post, i: number) =>
```

Comment:
If you have `!currentUser || currentUser.showDialoguesList`, you don't need the null-safe access after the second `currentUser`, since the expression will short-circuit in the first half if `currentUser` doesn't exist, and Typescript is smart enough to figure that out

Reply:
cool!